File: Aveline.Api/Modules/Commerce/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Aveline.Api.Common.MultiTenancy;
using Aveline.Api.Modules.Organizations.Models;
using Aveline.Api.Modules.Shared.Models;

namespace Aveline.Api.Modules.Commerce.Models;

[Table("Orders")]
public class Order : ITenantEntity
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    public Guid OrganizationId { get; set; }

    [Required]
    public Guid CustomerId { get; set; }

    [Required]
    [MaxLength(100)]
    public string CustomerName { get; set; } = string.Empty;

    [Required]
    [MaxLength(50)]
    public string OrderType { get; set; } = "whatsapp"; // in_store, whatsapp, instagram, sourcing

    [Required]
    [MaxLength(50)]
    public string Status { get; set; } = "pending_hold"; 
    // pending_hold, payment_requested, payment_confirmed, payment_expired, 
    // pending_approval, approved, rejected, delivery_scheduled, delivered, completed, cancelled

    [Column(TypeName = "decimal(18,2)")]
    public decimal Subtotal { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal Discount { get; set; } = 0.00m;

    [Column(TypeName = "decimal(18,2)")]
    public decimal Total { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal TotalCost { get; set; }

    [Column(TypeName = "decimal(5,4)")]
    public decimal Margin { get; set; } 

    public Guid? CreatedBy { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    // Navigation properties
    public Organization? Organization { get; set; }
    public User? CreatedByUser { get; set; }
    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
    public ICollection<Payment> Payments { get; set; } = new List<Payment>();
    public ICollection<ApprovalQueueEntry> Approvals { get; set; } = new List<ApprovalQueueEntry>();
    public DeliveryPlan? DeliveryPlan { get; set; }
}

File: frontend/aveline_mobile/lib/features/commerce/domain/entities/order.dart
import 'order_item.dart';

class Order {
  const Order({
    required this.id,
    required this.organizationId,
    required this.customerId,
    required this.customerName,
    required this.orderType,
    required this.status,
    required this.subtotal,
    required this.discount,
    required this.total,
    required this.totalCost,
    required this.margin,
    this.createdBy,
    required this.createdAt,
    this.updatedAt,
    this.items = const [],
  });

  final String id;
  final String organizationId;
  final String customerId;
  final String customerName;
  final String orderType; // in_store, whatsapp, instagram, sourcing
  final String status; // pending_approval, confirmed, processing, delivered, cancelled
  final double subtotal;
  final double discount;
  final double total;
  final double totalCost;
  final double margin;
  final String? createdBy;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final List<OrderItem> items;

  bool get isPendingApproval => status.toLowerCase() == 'pending_approval';
  bool get isConfirmed => status.toLowerCase() == 'confirmed';
  bool get isCancelled => status.toLowerCase() == 'cancelled';
  bool get isProcessing => status.toLowerCase() == 'processing';
  bool get isDelivered => status.toLowerCase() == 'delivered';

  /// The server's `margin` is a **ratio** in `[-1, 1]`, not a currency amount:
  /// `OrderService` computes `(total - totalCost) / total` (rounded to 4 dp).
  double get marginPercentage => margin * 100;

  /// The margin in currency: what is left of [total] after [totalCost].
  double get marginAmount => total - totalCost;
}
